### PR TITLE
Fix for making sure we get ResultWrapper from query().

### PR DESCRIPTION
**Pull request by harnash**
In some rare caese Database::query() returs false and it was causing
errors when trying to use reuturned value as an instance of ResultWrapper.
Added some safe checks to detect such cases.

### CODE REVIEW

File: extensions/wikia/ImageServing/drivers/ImageServingDriverMainNS.class.php [hunk 1]
```diff
@@ -137,17 +137,20 @@ protected function loadImageDetails( $imageNames = array() ) {
...
-						wfDebug( __METHOD__ . ": {$row->img_name} - filtered out because of {$row->img_minor_mime} minor MIME type\n" );
+			if ($result instanceof ResultWrapper && $result->numRows() > 0) {
+				foreach ($result as $row) {
+					/* @var mixed $row */
+					if ($row->img_height >= $this->minHeight && $row->img_width >= $this->minWidth) {
+						if (!in_array($row->img_minor_mime, self::$mimeTypesBlacklist)) {
+							$imageDetails[$row->img_name] = $row;
+						} else {
+							wfDebug(__METHOD__ . ": {$row->img_name} - filtered out because of {$row->img_minor_mime} minor MIME type\n");
+						}
```
Comment (macbre) [inline extensions/wikia/ImageServing/drivers/ImageServingDriverMainNS.class.php]: Your IDE got rid of all whitespaces that do follow MediaWiki / our conventions.

**Comment by michalroszka** [comment]
Does `false` indicate an error? Perhaps we should log it.

**Comment by macbre** [comment]
We actually do via `DBQueryError` exception handling.

**Comment by michalroszka** [comment]
@macbre but for some reason we still end up with `loadImageDetails` causing errors?

File: extensions/wikia/ImageServing/drivers/ImageServingDriverMainNS.class.php (hunk 1 above)
Comment (harnash) [inline extensions/wikia/ImageServing/drivers/ImageServingDriverMainNS.class.php]: Fixed in: 7a3c6506bbf31faf9c9037495e0f37aca689f346

**Comment by macbre** [comment]
LGTM

File: extensions/wikia/ImageServing/drivers/ImageServingDriverMainNS.class.php [hunk 2]
```diff
@@ -137,17 +137,20 @@ protected function loadImageDetails( $imageNames = array() ) {
...
 			);
 
-			foreach ( $result as $row ) {
-				/* @var mixed $row */
-				if ( $row->img_height >= $this->minHeight && $row->img_width >= $this->minWidth ) {
-					if ( !in_array( $row->img_minor_mime, self::$mimeTypesBlacklist ) ) {
-						$imageDetails[$row->img_name] = $row;
-					} else {
-						wfDebug( __METHOD__ . ": {$row->img_name} - filtered out because of {$row->img_minor_mime} minor MIME type\n" );
+			if ( $result instanceof ResultWrapper && $result->numRows() > 0 ) {
```
Comment (wladekb) [inline extensions/wikia/ImageServing/drivers/ImageServingDriverMainNS.class.php]: with ->numRows() == 0 we probably still want to call ->free()

**Comment by wladekb** [comment]
->selectField() can also return false when query succeeded but no row was found

File: extensions/wikia/ImageServing/drivers/ImageServingDriverMainNS.class.php (hunk 2 above)
Comment (harnash) [inline extensions/wikia/ImageServing/drivers/ImageServingDriverMainNS.class.php]: Fixed in  8b169f0

**Pull request merged**